pkg/compose/restart.go

```diff
@@ -48,10 +48,10 @@ func (s *composeService) restart(ctx context.Context, projectName string, option
 		}
 	}
 
-	// ignore depends_on relations which are not impacted by restarting service
+	// ignore depends_on relations which are not impacted by restarting service or not required
 	for i, service := range project.Services {
 		for name, r := range service.DependsOn {
-			if !r.Restart {
+			if !r.Restart || !r.Required {
```

Comment:
Similar to @ndeloof's [Q around wait](https://github.com/docker/compose/pull/10792/files#r1266706274), should this not filter them out, but ignore any failures attempting to restart IF it was not required?